src/sort/mod.rs

```diff
@@ -56,6 +56,13 @@ pub trait Sort: Any + Send + Sync + Debug {
         false
     }
 
+    /// Return the inner values and sorts.
```

Comment:
At first I had re-written `foreach_tracked_values` to use this new function, but then reverted to not degrade that performance at all.

Reply:
In a small test on my computer, rewriting `foreach_tracked_values` to use this function didn't slow things down, so I changed it to do that again, to reduce code duplication.